packages/cosmic-swingset/src/extract-proposal.js

```diff
@@ -97,25 +96,60 @@ export const extractCoreProposalBundles = async (
         );
         return handle;
       };
-      const proposal = await deeplyFulfilled(
-        harden(ns.defaultProposalBuilder({ publishRef, install })),
-      );
-      const { sourceSpec, getManifestCall } = proposal;
+      const proposal = await ns.defaultProposalBuilder({ publishRef, install });
+
+      const bundleSpecs = [];
+      // Add the proposal bundle handles in sorted order.
+      [...bundleHandles.values()]
+        .sort(({ source: a }, { source: b }) => {
+          if (a < b) {
+            return -1;
+          }
+          if (a > b) {
+            return 1;
+          }
+          return 0;
+        })
+        .forEach((handle, j) => {
+          handle.bundleID = `coreProposal${i}_${j}`;
+          harden(handle);
+          bundleSpecs.push([handle.bundleID, { sourceSpec: handle.source }]);
+        });
 
+      // Now that we've assigned all the bundleIDs and hardened the handles, we
+      // can extract the behavior bundle.
+      const { sourceSpec, getManifestCall } = await deeplyFulfilled(
+        harden(proposal),
+      );
```

Comment:
I find it unfortunate we need to jump through hoops because `deeplyFulfilled` automatically `harden`s.

Reply:
It's not that `deeplyFulfilled` hardens, it's that it requires hardened input.  But yes, it's unfortunate.